consensus/messages.go

```diff
@@ -42,54 +41,60 @@ func CreateProposeMessage(
 }
 
 func CreateVoteMessage(
-	m *ConsensusModule,
-	step typesCons.HotstuffStep, // step can be taken from `m` but is specified explicitly via interface to avoid ambiguity
+	height uint64,
+	round uint64,
+	step typesCons.HotstuffStep,
 	block *typesCons.Block,
+	privKey crypto.PrivateKey, // used to sign the vote
 ) (*typesCons.HotstuffMessage, error) {
 	if block == nil {
 		return nil, typesCons.ErrNilBlockVote
 	}
 
 	msg := &typesCons.HotstuffMessage{
 		Type:          Vote,
-		Height:        m.Height,
+		Height:        height,
 		Step:          step,
-		Round:         m.Round,
+		Round:         round,
 		Block:         block,
 		Justification: nil, // signature is computed below
 	}
 
 	msg.Justification = &typesCons.HotstuffMessage_PartialSignature{
 		PartialSignature: &typesCons.PartialSignature{
-			Signature: getMessageSignature(msg, m.privateKey),
-			Address:   m.privateKey.PublicKey().Address().String(),
+			Signature: getMessageSignature(msg, privKey),
+			Address:   privKey.PublicKey().Address().String(),
 		},
 	}
 
 	return msg, nil
 }
 
-// Returns a "partial" signature of the hotstuff message from one of the validators
-func getMessageSignature(m *typesCons.HotstuffMessage, privKey crypto.PrivateKey) []byte {
-	bytesToSign, err := getSignableBytes(m)
+// Returns "partial" signature of the hotstuff message from one of the validators.
+// If there is an error signing the bytes, nil is returned instead.
+func getMessageSignature(msg *typesCons.HotstuffMessage, privKey crypto.PrivateKey) []byte {
+	bytesToSign, err := getSignableBytes(msg)
 	if err != nil {
+		log.Printf("[WARN] error getting bytes to sign: %v\n", err)
 		return nil
 	}
+
 	signature, err := privKey.Sign(bytesToSign)
 	if err != nil {
-		log.Fatalf("Error signing message: %v", err)
+		log.Printf("[WARN] error signing message: %v\n", err)
 		return nil
 	}
+
 	return signature
 }
 
-// Signature should only be over a subset of the fields in a HotstuffMessage
-func getSignableBytes(m *typesCons.HotstuffMessage) ([]byte, error) {
+// Signature only over subset of fields in HotstuffMessage
```

Comment:
noop: I am sure there's a reason but I am gonna ask why offline. 

What happens if I am a node that signs and sends different blocks with different justifications?

Reply:
Added this to the comment:
```
// For reference, see section 4.3 of the the hotstuff whitepaper, partial signatures are
// computed over `tsignr(hm.type, m.viewNumber , m.nodei)`. https://arxiv.org/pdf/1803.05069.pdf
```

Does this make more sense?
- Same block
- It's a partial signature by 1 node

Reply:
Makes sense, thank you!